src/custom/state/claim/hooks/index.ts

```diff
@@ -1061,3 +1066,71 @@ export function useVCowData(): VCowData {
 
   return { isLoading, vested, unvested, total }
 }
+
+/**
+ * Hook used to swap vCow to Cow token
+ */
+
+interface SwapVCowCallbackParams {
+  openModal: (message: string, operationType: OperationType) => void
+  closeModal: () => void
+}
+
+export function useSwapVCowCallback({ openModal, closeModal }: SwapVCowCallbackParams) {
+  const { chainId, account } = useActiveWeb3React()
+  const vCowContract = useVCowContract()
+
+  const addTransaction = useTransactionAdder()
+  const vCowToken = chainId ? V_COW[chainId] : undefined
+
+  const swapCallback = useCallback(async () => {
+    if (!account) {
+      throw new Error('Not connected')
+    }
+    if (!chainId) {
+      throw new Error('No chainId')
+    }
+    if (!vCowContract) {
+      throw new Error('vCOW contract not present')
+    }
+    if (!vCowToken) {
+      throw new Error('vCOW token not present')
+    }
+
+    const args: { from: string; gasLimit?: BigNumber } = {
+      from: account,
+    }
+
+    const estimatedGas = await vCowContract.estimateGas.swapAll(args).catch(() => {
+      // general fallback for tokens who restrict approval amounts
+      return vCowContract.estimateGas.swapAll().catch((error) => {
+        console.log(
+          '[useSwapVCowCallback] Error estimating gas for swapAll. Using default gas limit ' +
+            APPROVE_GAS_LIMIT_DEFAULT.toString(),
+          error
+        )
+        return APPROVE_GAS_LIMIT_DEFAULT
+      })
+    })
+
+    args.gasLimit = estimatedGas
```

Comment:
i find this more declarative if you don't even declare the args and then add the estimation here, if you just create the object in 1122. This way is easier to read

Reply:
Updated